src/battle-log.ts

```diff
@@ -126,7 +126,11 @@ class BattleLog {
 			return;
 
 		case 'unlink':
-			this.hideChatFrom(toId(args[2] || args[1]));
+			const user = toId(args[2]) || toId(args[1]);
+			this.unlinkChatFrom(user);
```

Comment:
This shouldn't be done if `args[2]` exists.

Reply:
Since the whole thing is |unlink|hide|, I assumed it wasn't a bad idea to actually unlink _and_ hide, since I see no reason to leave links intact after they've been hidden like that (in case someone else unhides the messages)